src/sys/socket/sockopt.rs

```diff
@@ -987,6 +987,25 @@ sockopt_impl!(
     libc::SO_ACCEPTFILTER,
     libc::accept_filter_arg
 );
+#[cfg(target_os = "illumos")]
+sockopt_impl!(
+    /// Attach a named filter to this socket to be able to
+    /// defer when anough byte had been buffered by the kernel
+    FilAttach,
+    SetOnly,
+    libc::SOL_FILTER,
+    libc::FIL_ATTACH,
+    SetOsString<'static>
```

Comment:
I am thinking if this `'static` lifetime would be too restrictive? I do not see a way to specify lifetime parameter using the macro, maybe we can implement the `nix::sys::socket::SetSockOpt` trait manually?